backend/src/apiserver/common/config.go

```diff
@@ -30,8 +30,13 @@ const (
 	DefaultPipelineRunnerServiceAccount string = "DefaultPipelineRunnerServiceAccount"
 	KubeflowUserIDHeader                string = "KUBEFLOW_USERID_HEADER"
 	KubeflowUserIDPrefix                string = "KUBEFLOW_USERID_PREFIX"
+	UpdatePipelineVersionByDefault      string = "UpdatePipelineVersionByDefault"
```

Comment:
nit: AUTO_UPDATE_PIPELINE_DEFAULT_VERSION?

this may be specified from env, and there is no casing there, so prefer _

You can adjust the wording, I just thought update pipeline default version by default is too long

Reply:
I will fix it.

Reply:
So as I understand it we will move away from using the [config.go](https://github.com/kubeflow/pipelines/blob/master/backend/src/apiserver/common/const.go) file for configurations. Instead I update the param.env and make sure it is available as a env variable in the ml-pipeline-apiserver-deployment.yaml [deployment](https://github.com/kubeflow/pipelines/blob/d91a0c9da12bbc22019e4df49314387f6052ad18/manifests/kustomize/base/pipeline/ml-pipeline-apiserver-deployment.yaml#L17).   And collect the value as an env variable in the code ?

Reply:
No, in fact the viper library in config.go gets config from multiple sources with priority order.

For example: https://github.com/kubeflow/pipelines/blob/c32ea232d5319728229fb2439f795dcc013be7f7/backend/src/apiserver/common/config.go#L100
it can be specified by both env and config.json. While env has higher priority.

Reply:
ohhh cool, then I will revert back and update.